paddle/fluid/distributed/auto_parallel/spmd_rules/layer_norm_spmd_rule.cc

```diff
@@ -127,8 +126,8 @@ LayerNormSPMDRule::InferForward(const std::vector<DistTensorSpec>& input_specs,
   for (size_t i = 0; i < out_axes.size(); ++i) {
     if (i < static_cast<size_t>(begin_norm_axis)) {
       out_dims_mapping.push_back(x_dims_mapping[i]);
-      // if ijk,k,k->ijk,x,x (x,scale,bias->out,mean,variance,
-      // begin_norm_axis=2, x=ij), and the dims_mapping of input is (0,1,-1),
+      // if ijk,k,k->ijk,z,z (x,scale,bias->out,mean,variance,
```

Comment:
the current rule for LN is problematic:

- not all axes before begin_norm_axis could be sharded.
- the first axis after begin_norm_axis could sharded in current implementation.
- it would be better to refactor the LN rule using TransDim algorithms: the axes mapping is like axes-flatten in TransDim.

Reply:
but it is ok for now, since most usage of LN is DP.

Reply:
Done, now only the first axis of the input can be sharded, other axes will be set to replicated.